### ISSUE TITLE
How to have some labels ignored in the Dice loss computation during training?

### ISSUE THREAD

**Issue by RomStriker** [MIC-DKFZ/nnUNet#570]
Hi,

I have some labels for my data that are neither background nor foreground. I would like these to be ignored during training. How do I do that with nnUNet? Thanks.

**Comment by FabianIsensee** [comment]
Hi, this is not yet implemented. We are planning to implement this in the future.
Best,
Fabian

**Comment by RomStriker** [comment]
Thank you. Can you point me to the functions that I can modify to make this happen? I tried searching for all the loss functions used, but there are quite a few. Which ones do you think I need to modify for the network to ignore the ignored labels completely during training.

**Comment by FabianIsensee** [comment]
I mean - nnU-Net uses CE + Dice loss. So those are the two you need to be looking at ;-)

**Comment by FabianIsensee** [comment]
(SoftDiceLoss)

**Comment by RomStriker** [comment]
Alright, thanks. I'll check what I can do.

**Comment by Nora-Go** [comment]
Just a side question: Isn't the softDiceLoss = 1 - Dice Coefficient? You seem to be using simply using the negative Dice Coefficient (which is of course totally fine as well, just confused me at first to get negative losses)

**Comment by FabianIsensee** [comment]
`1 - ` makes no sense even though everyone seems to be using this. 1 is just a constant and doesn't play a role in gradient computation.
SoftDiceLoss is not exactly the same as Dice. It does not operate on 'hard' predictions and instead uses softmax outputs. Also keep in mind that some people (though not us) use a square term in the computation. 
Best,
Fabian

**Comment by Nora-Go** [comment]
ah thanks! :D I guess the 1 is for people like me who are confused seeing negative losses. But the more I know! Great work by the way ;)

**Comment by FabianIsensee** [comment]
Yeah some people get confused by the loss value :-) Too many forget that the value of the loss doesn't matter in most cases. Only the gradients do
